contrib/TestHarness2/test_harness/joshua.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import collections
+import io
+import sys
+import xml.sax
+import xml.sax.handler
+from pathlib import Path
+from typing import List, OrderedDict, Set
+
+from joshua import joshua_model
```

Comment:
It looks like this depends on joshua, which is possibly worth calling out somehow. This will be part of joshua's de facto interface

Reply:
`test_harness` doesn't really depend on joshua -- the higher-level import is enclosed in a `try-except` clause (and an exception won't make it fail). However, if joshua is available it will also print a full summery of the failures.

This tool certainly needs some documentation, but I was planning to do this in a later PR